agent/src/test/java/com/code_intelligence/jazzer/api/AutofuzzTest.java

```diff
@@ -42,4 +59,49 @@ public void testConsumeFailsWithoutException() {
         (byte) 1 /* do not return null without searching for implementing classes */));
     assertNull(Jazzer.consume(data, UnimplementedInterface.class));
   }
+
+  @Test
+  public void testAutofuzz() {
+    FuzzedDataProvider data = CannedFuzzedDataProvider.create(
+        Arrays.asList((byte) 1 /* do not return null */, 0 /* first class on the classpath */,
+            (byte) 1 /* do not return null */, 0 /* first constructor */));
+    assertEquals(Boolean.TRUE,
+        Jazzer.autofuzz(data, (Function1<ImplementedInterface, ?>) AutofuzzTest::implIsNotNull));
+  }
+
+  @Test
+  public void testAutofuzzFailsWithException() {
+    FuzzedDataProvider data = CannedFuzzedDataProvider.create(
+        Collections.singletonList((byte) 1 /* do not return null */));
+    try {
+      Jazzer.autofuzz(data, (Function1<UnimplementedInterface, ?>) AutofuzzTest::implIsNotNull);
+    } catch (AutofuzzConstructionException e) {
+      // Pass.
+      return;
+    }
+    fail("should have thrown an AutofuzzConstructionException");
+  }
+
+  @Test
+  public void testAutofuzzConsumer() {
+    FuzzedDataProvider data = CannedFuzzedDataProvider.create(
+        Arrays.asList((byte) 1 /* do not return null */, 6 /* string length */, "foobar", 42,
+            (byte) 5, (byte) 1 /* do not return null */, 0 /* first class on the classpath */,
+            (byte) 1 /* do not return null */, 0 /* first constructor */));
+    Jazzer.autofuzz(data, AutofuzzTest::checkAllTheArguments);
+  }
+
+  @Test
+  public void testAutofuzzConsumerThrowsException() {
+    FuzzedDataProvider data =
+        CannedFuzzedDataProvider.create(Arrays.asList((byte) 1 /* do not return null */,
+            6 /* string length */, "foobar", 42, (byte) 5, (byte) 0 /* *do* return null */));
```

Comment:
the last 0 is to return `null`, right?

Reply:
Yes, that is also why I left the comment with an emphasis on `do`. Is that clear enough?